address-controller/src/main/java/io/enmasse/controller/standard/AddressController.java

```diff
@@ -116,21 +116,26 @@ public synchronized void resourcesUpdated(Set<Address> newAddressSet) throws Exc
             addressByGroup.get(key).add(address);
         }
 
-        validateAddressGroups(addressByGroup);
-
-        List<AddressCluster> clusterList = kubernetes.listClusters();
-        log.debug("Current set of clusters: " + clusterList);
-        deleteBrokers(clusterList, addressByGroup);
-        createBrokers(clusterList, addressByGroup);
-
-        // Perform status check
-        checkStatuses(newAddressSet);
-        for (Address address : newAddressSet) {
-            try {
-                addressApi.replaceAddress(address);
-            } catch (KubernetesClientException ex) {
-                log.warn("Error replacing address {}", address, ex);
+        try {
+            validateAddressGroups(addressByGroup);
+
+            List<AddressCluster> clusterList = kubernetes.listClusters();
+            log.debug("Current set of clusters: " + clusterList);
+            deleteBrokers(clusterList, addressByGroup);
+            createBrokers(clusterList, addressByGroup);
+
+            // Perform status check
+            checkStatuses(newAddressSet);
+            for (Address address : newAddressSet) {
+                try {
+                    addressApi.replaceAddress(address);
+                } catch (KubernetesClientException ex) {
+                    log.warn("Error syncing address {}", address, ex);
+                    eventLogger.log("AddressSyncFailed", "Error syncing address: " + ex.getMessage(), "Error", "Address", address.getName());
```

Comment:
https://kubernetes.io/docs/api-reference/v1.8/#event-v1-core only talks of event types "Normal" and "Warning", is "Error" also valid (it seems perfectly sensible for it to be so, but the link only talks of "new types could be added in the future"